teps/0092-scheduling-timeout.md

```diff
@@ -17,361 +24,474 @@ authors:
-  both timeouts has been counted, e.g. not just the `execution` timeout
+
+### PipelineRun
+
+#### TaskRunTemplate
+
+In [TEP-0119][tep-0119], we added `TaskRunTemplate` to support setting configuration for all `TaskRun` in `PipelineRun`.  This proposal adds a new timeout configuration within a `TaskRunTemplate` that allows users to specify `scheduling`, `execution` and `total` timeout for all `TaskRuns` executed in a given `PipelineRun`.
+
+```yaml=
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: clone-test-build-
+spec:
+  pipelineRef:
+    name: clone-test-build
+  taskRunTemplate:
+    timeouts:
+      scheduling: "10m"
+      execution: "30m"
+      total: "40m"
+```
+
+#### TaskRunSpec
+
+In [TEP-0119][tep-0119], we added `TaskRunSpec` to support setting configuration for a specific `TaskRun` in `PipelineRun`. This proposal adds a new timeout configuration within a `TaskRunSpec` that allows users to specify a `scheduling`, `execution` and `total` timeout for a specific `TaskRun` using `TaskRunSpecs`.
+
+```yaml=
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: clone-test-build-
+spec:
+  pipelineRef:
+    name: clone-test-build
+  taskRunSpecs:
+    - pipelineTaskName: clone
+      timeouts:
+        scheduling: "10m"
+        execution: "30m"
+        total: "40m"
+```
+
+All the `TaskRuns` would have the specified timeouts applied.
+
+#### TaskRunTemplate and TaskRunSpec
+
+When the `TaskRunTemplate` is specified with timeouts for all `TaskRuns`, a user can override the timeouts for a specific `TaskRun` using `TaskRunSpecs`.
+
+```yaml=
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: clone-test-build-
+spec:
+  pipelineRef:
+    name: clone-test-build
+  taskRunTemplate:
+    timeouts:
+      scheduling: "10m"
+      execution: "30m"
+      total: "40m"
+  taskRunSpecs:
+    - pipelineTaskName: build
+      timeouts:
+        scheduling: "20m"
+        execution: "40m"
+        total: "60m"
+```
+
+The "clone" and "test" `TaskRuns` would have the timeouts in the `TaskRunTemplate`, while "build" `TaskRun` would have its timeout that's specified in the `TaskRunSpecs`.
+
+
+### Pipeline
+
+The existing `timeout` field in `Pipelines`  allows users to specify the timeout for a specific `TaskRun` executed from a `Pipeline`. That is, [`pipeline.pipelinetask.timeout`][pipeline-timeout] is passed to the `taskrun.timeout`.
+
+```yaml=
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-test-build
+spec:
+  tasks:
+  - name: clone
+    timeout: 10m
+    taskRef:
+      name: git-clone
+  - name: test
+    timeout: 5m
+    runAfter:
+    - clone
+    taskRef:
+      name: junit
+```
+
+Given that `taskrun.timeouts.total` is replacing `taskrun.timeout`, the existing `pipeline.pipelinetask.timeout` will now be passed to the new `taskrun.timeouts.total`.
```

Comment:
Do we plan to allow to specify scheduling and execution timeouts in there as well ?